src/ape/pytest/fixtures.py

```diff
@@ -20,3 +24,78 @@ def chain(self) -> ChainManager:
     @pytest.fixture(scope="session")
     def project(self) -> ProjectManager:
         return self.project_manager
+
+    @pytest.fixture(scope="session")
+    def _session_isolation(self) -> Iterator[None]:
+        snapshot_id = None
+        try:
+            snapshot_id = self.chain_manager.snapshot()
+        except NotImplementedError:
+            self._warn_for_unimplemented_snapshot()
+
+        yield
+
+        if snapshot_id is not None and snapshot_id in self.chain_manager._snapshots:
+            self.chain_manager.restore(snapshot_id)
+
+    @pytest.fixture(scope="package")
+    def _package_isolation(self, _session_isolation) -> Iterator[None]:
+        snapshot_id = None
+        try:
+            snapshot_id = self.chain_manager.snapshot()
+        except NotImplementedError:
+            self._warn_for_unimplemented_snapshot()
+
+        yield
+
+        if snapshot_id is not None and snapshot_id in self.chain_manager._snapshots:
+            self.chain_manager.restore(snapshot_id)
+
+    @pytest.fixture(scope="module")
+    def _module_isolation(self, _package_isolation) -> Iterator[None]:
+        snapshot_id = None
+        try:
+            snapshot_id = self.chain_manager.snapshot()
+        except NotImplementedError:
+            self._warn_for_unimplemented_snapshot()
+
+        yield
+
+        if snapshot_id is not None and snapshot_id in self.chain_manager._snapshots:
+            self.chain_manager.restore(snapshot_id)
+
+    @pytest.fixture(scope="class")
+    def _class_isolation(self, _module_isolation) -> Iterator[None]:
+        snapshot_id = None
+        try:
+            snapshot_id = self.chain_manager.snapshot()
+        except NotImplementedError:
+            self._warn_for_unimplemented_snapshot()
+
+        yield
+
+        if snapshot_id is not None and snapshot_id in self.chain_manager._snapshots:
+            self.chain_manager.restore(snapshot_id)
+
+    @pytest.fixture(scope="function")
+    def _function_isolation(self, _class_isolation) -> Iterator[None]:
+        snapshot_id = None
+        try:
+            snapshot_id = self.chain_manager.snapshot()
+        except NotImplementedError:
+            self._warn_for_unimplemented_snapshot()
+
+        yield
+
+        if snapshot_id is not None and snapshot_id in self.chain_manager._snapshots:
+            self.chain_manager.restore(snapshot_id)
+
+    def _warn_for_unimplemented_snapshot(self):
+        if self._warned_for_missing_features:
+            return
+
+        logger.warning(
```

Comment:
Would this warning still occur if the user said --disable-isolation and was using a provider that didnt support snapshotting?

I think it shouldn't! But can't really tell, I am reviewing from
my phone so I can't test it out right now

Reply:
Definitely won't, since `--disable-isolation` doesn't inject the fixtures, this wouldn't get called.